index.js.flow

```diff
@@ -57,11 +57,15 @@ type AssertContext = {
 	notDeepEqual<U>(value: U, expected: U, message?: string): void;
 	// Assert that function throws an error or promise rejects.
 	// @param error Can be a constructor, regex, error message or validation function.
-	throws(value: PromiseLike<mixed>, error?: ErrorValidator, message?: string): Promise<mixed>;
-	throws(value: () => void, error?: ErrorValidator, message?: string): mixed;
+	throws: {
+		(value: PromiseLike<mixed>, error?: ErrorValidator, message?: string): Promise<void>;
+		(value: () => mixed, error?: ErrorValidator, message?: string): void;
```

Comment:
> Specifically, ava doesn't care what values these throwing functions return, but doesn't produce a return value.

`t.throws()` returns the error (and throws an assertion error if no error was thrown). Perhaps we could do something like:

```typescript
throws<E>: {
  		(value: PromiseLike<mixed>, error?: ErrorValidator, message?: string): Promise<E>;
  		(value: () => mixed, error?: ErrorValidator, message?: string): E;
}
```

I don't really know how generics work in Flow though. And of course we wouldn't want to require people to specify the error type if they're not going to use it.